src/shared/components/axisTicks/AxisTicksGenerator.tsx

```diff
@@ -0,0 +1,203 @@
+// Libraries
+import React, {FC, useState} from 'react'
+import {connect, ConnectedProps} from 'react-redux'
+import {get} from 'lodash'
+
+// Utils
+import {isFlagEnabled} from 'src/shared/utils/featureFlag'
+import {getActiveTimeMachine} from 'src/timeMachine/selectors'
+import {capitalize} from 'lodash'
+
+// Components
+import {ValueTicksOptions} from 'src/shared/components/axisTicks/ValueTicksOptions'
+import {TimeTicksOptions} from 'src/shared/components/axisTicks/TimeTicksOptions'
+import {Columns, Form, Grid, SelectGroup} from '@influxdata/clockface'
+
+// Types
+import {AppState} from 'src/types'
+import {ButtonShape} from '@influxdata/clockface'
+
+const getGenerateAxisTicksOptionsNames = (axisName: string) => {
+  const result = []
+  result.push(`generate${axisName.toUpperCase()}AxisTicks`)
+  result.push(`${axisName.toLowerCase()}TotalTicks`)
+  result.push(`${axisName.toLowerCase()}TickStart`)
+  result.push(`${axisName.toLowerCase()}TickStep`)
+  return result
+}
+
+interface AxisTicksGeneratorProps {
+  axisName: string
+  columnType: string
+  onSetGenerateAxisTicks: (generateAxisTicks: string[]) => void
+  onSetTotalTicks: (totalTicks: number) => void
+  onSetTickStart: (tickStart: number) => void
+  onSetTickStep: (tickStep: number) => void
+  label?: string
+}
+
+type ReduxProps = ConnectedProps<typeof connector>
+type Props = AxisTicksGeneratorProps & ReduxProps
+
+const AxisTicksGenerator: FC<Props> = props => {
+  const {
+    axisName,
+    columnType,
+    generateAxisTicks,
```

Comment:
yeah, it does sound like a function. However, "generated" is not correct either. Because it really is just a list of options to use for generating the ticks, not the ticks themselves.

Reply:
I wanted to match the name in the api, so that's how it ended up as "generateAxisTicks".

The bigger issue is that it might be misnamed in the api as well, if "generateAxisTicks" sounds too much like a function.